src/goto-programs/character_refine_preprocess.cpp

```diff
@@ -0,0 +1,2449 @@
+          corresponding to the java method
+          Character.isUnicodeIdentifierStart:(I)Z
+
+\*******************************************************************/
+
+void character_refine_preprocesst::convert_is_unicode_identifier_start_int(
+  conversion_input &target)
+{
+  convert_is_unicode_identifier_start_char(target);
+}
+
+/*******************************************************************\
+
+Function: character_refine_preprocesst::convert_is_upper_case_char
+
+  Inputs:
+    target - a position in a goto program
+
+ Purpose: Converts function call to an assignement of an expression
+          corresponding to the java method Character.isUpperCase:(C)Z
+
+    TODO: For now we only consider ASCII characters
+
+\*******************************************************************/
+
+void character_refine_preprocesst::convert_is_upper_case_char(
+  conversion_input &target)
+{
+  convert_char_function(
+    &character_refine_preprocesst::expr_of_is_ascii_upper_case, target);
+}
+
+/*******************************************************************\
+
+Function: character_refine_preprocesst::convert_is_upper_case_int
+
+  Inputs:
+    target - a position in a goto program
+
+ Purpose: Converts function call to an assignement of an expression
+          corresponding to the java method Character.isUpperCase:(I)Z
+
+\*******************************************************************/
+
+void character_refine_preprocesst::convert_is_upper_case_int(
+  conversion_input &target)
+{
+  convert_is_upper_case_char(target);
+}
+
+/*******************************************************************\
+
+Function: character_refine_preprocesst::convert_is_valid_code_point
+
+  Inputs:
+    target - a position in a goto program
+
+ Purpose: Converts function call to an assignement of an expression
+          corresponding to the java method Character.isValidCodePoint:(I)Z
+
+\*******************************************************************/
+
+void character_refine_preprocesst::convert_is_valid_code_point(
+    conversion_input &target)
+{
+  // TODO
+}
+
+/*******************************************************************\
+
+Function: character_refine_preprocesst::expr_of_is_whitespace
+
+  Inputs:
+    expr - An expression of type character
+    type - A type for the output
+
+ Outputs: A Boolean expression
+
+ Purpose: Determines if the specified character is white space according
+          to Java. It is the case when it one of the following:
+          * a Unicode space character (SPACE_SEPARATOR, LINE_SEPARATOR, or
+            PARAGRAPH_SEPARATOR) but is not also a non-breaking space
+            ('\u00A0', '\u2007', '\u202F').
+          * it is one of these: U+0009  U+000A U+000B U+000C U+000D
+            U+001C U+001D U+001E U+001F
+
+\*******************************************************************/
+
+exprt character_refine_preprocesst::expr_of_is_whitespace(
+  exprt expr, typet type)
```

Comment:
Rather random as it applies across this PR: use (const) references wherever possible, both in parameters and within the functions.